app/registries/foreman/plugin.rb

```diff
@@ -346,12 +354,13 @@ def add_all_permissions_to_default_roles
 
     def pending_migrations
       return true if Foreman.in_setup_db_rake?
+      return @pending_migrations unless @pending_migrations.nil?
 
-      pending_migrations = ActiveRecord::Base.connection.migration_context.needs_migration?
+      @pending_migrations = ActiveRecord::MigrationContext.new(migrations_paths, ActiveRecord::SchemaMigration).needs_migration?
 
-      Rails.logger.debug("There are pending migrations. Please run foreman-rake db:migrate.") if pending_migrations
+      Rails.logger.debug("There are pending migrations. Please run foreman-rake db:migrate.") if @pending_migrations
 
-      pending_migrations
+      @pending_migrations
     end
```

Comment:
Isn't there something like:
```ruby
@pending_migrations ||= begin
  if ActiveRecord::MigrationContext.new(migrations_paths, ActiveRecord::SchemaMigration).needs_migration?
    Rails.logger.debug("There are pending migrations for #{id}. Please run foreman-rake db:migrate.")
    true
  else
    false
  end
end
```

Reply:
Oh, disregard that since `@pending_migrations` is already short circuited earlier and wouldn't work due to false/nil.

Reply:
yes, but I like the idea of stating for which plugin the migrations are pending. Let me add it.